### PR TITLE
Support C# 12 by installing .NET 8 SDK in the Ubuntu pipeline

### PR DESCRIPTION
**Pull request by Bartleby2718**
### Motivation
The current `Ubuntu` pipeline doesn't have .NET 8 SDK installed.

### Background
According to https://www.appveyor.com/docs/linux-images-software/, we're currently using Ubuntu 18.04 (image name: `Ubuntu`) in the CI pipeline.

However, .NET 8 SDK is not installed in any of default Ubuntu images according to the same page. Until https://github.com/appveyor/ci/issues/3908 is complete, we'll need to install .NET SDK manually.

However, you cannot install .NET 8 on Ubuntu 18.04 per https://learn.microsoft.com/en-us/dotnet/core/install/linux-ubuntu-1804.

### Changes in the PR
Updated the image to the latest Ubuntu version (that's supported in AppVeyor) 22.04 and manually installing .NET 8 SDK per https://learn.microsoft.com/en-us/dotnet/core/install/linux-ubuntu-2204.

### Testing
- I included C# 12 changes like primary constructors and collection expressions, which come with .NET 8 SDK, so this should be a good litmus test.
- The Windows build is failing due to #110.

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -2,7 +2,12 @@ version: 1.0.{build}
 
 image: 
   - Visual Studio 2022
-  - Ubuntu
+  - Ubuntu2204
+
+init:
```
Comment (Bartleby2718) [inline appveyor.yml]: Per https://www.appveyor.com/docs/build-configuration/#build-pipeline `init` runs first.

File: appveyor.yml [hunk 2]
```diff
@@ -2,7 +2,12 @@ version: 1.0.{build}
 
 image: 
   - Visual Studio 2022
-  - Ubuntu
+  - Ubuntu2204
```
Comment (Bartleby2718) [inline appveyor.yml:5]: Latest in https://www.appveyor.com/docs/linux-images-software/

File: appveyor.yml [hunk 3]
```diff
@@ -2,7 +2,12 @@ version: 1.0.{build}
...
+
+init:
+  - sh: |
```
Comment (Bartleby2718) [inline appveyor.yml]: 1. `sh:`

https://www.appveyor.com/docs/getting-started-with-appveyor-for-linux/#bash-and-powershell
> To run Bash command either put it with `sh:` prefix

2. `|`

One of the [many ways YAML does multi-line strings](https://stackoverflow.com/questions/3790454/how-do-i-break-a-string-in-yaml-over-multiple-lines) (literal style). Didn't think too much about this, so happy to change this to folded style.

File: appveyor.yml [hunk 4]
```diff
@@ -2,7 +2,12 @@ version: 1.0.{build}
 
 image: 
   - Visual Studio 2022
-  - Ubuntu
+  - Ubuntu2204
+
+init:
+  - sh: >
```
Comment (Bartleby2718) [inline appveyor.yml]: 1. `sh:`

https://www.appveyor.com/docs/getting-started-with-appveyor-for-linux/#bash-and-powershell
> To run Bash command either put it with `sh:` prefix

2. `>`

One of the [many ways YAML does multi-line strings](https://stackoverflow.com/questions/3790454/how-do-i-break-a-string-in-yaml-over-multiple-lines) (folded style).

**Comment by Bartleby2718** [comment]
@madelson No transient failures this time!

**Review by madelson — APPROVED** [review]

**Pull request merged**